### PR TITLE
[REF] Move ParticipantCount into a JS file, remove eval(), whitespace cleanup

### PR DESCRIPTION
**Pull request by MegaphoneJon**
Overview
----------------------------------------
I tried fixing a bug in #20800 but that file was cursed.  I don't really care about the bugfix that much, so I decided my alternate good deed would be to clean the file.

Before
----------------------------------------
JS in a .tpl file
Code formatting linters ran away screaming
use of `eval()`.
File only loaded if waitlist is enabled.

After
----------------------------------------
JS in a .js file
Consistent code formatting
Replace `eval()` with `JSON.parse()`.
File loaded if multiple registrations and maximum participants are enabled.

### CODE REVIEW

File: templates/CRM/Price/Form/ParticipantCount.tpl [hunk 1]
```diff
@@ -1,191 +0,0 @@
...
- +--------------------------------------------------------------------+
- | Copyright CiviCRM LLC. All rights reserved.                        |
- |                                                                    |
- | This work is published under the GNU AGPLv3 license with some      |
- | permitted exceptions and without any warranty. For full license    |
- | and copyright information, see https://civicrm.org/licensing       |
- +--------------------------------------------------------------------+
-*}
-
-{if $priceSet && $allowGroupOnWaitlist}
```
Comment (demeritcowboy) [inline templates/CRM/Price/Form/ParticipantCount.tpl:11]: Civi is full of fun stuff to find, huh?

I'm not immediately seeing what this smarty `{if}` was replaced by? There's `if ($this->_values['event']['is_multiple_registrations'] && isset($this->_values['event']['max_participants']))` but it doesn't look like the same thing?

Also I see two other templates that will still try to include this tpl?

Comment (MegaphoneJon) [reply templates/CRM/Price/Form/ParticipantCount.tpl:11]: Hmm - I assumed that was the only screen that would have any reason to include this .tpl, and I relied on the change I made to the PHP file that determined whether to load this .tpl at all.  I'll go back and review the other files that include this.

**Comment by MegaphoneJon** [comment]
Oh right - I remember now - I worked on this in preference to #20800 because of the aforementioned curse.

Originally I needed the `pricesetParticipantCount()` for reasons unrelated to waitlist - I have an extension that improves the UX for multiple-participant registrations.  Which is why I changed the conditional (I mention it up top).  

I'm trying to remember what my motivation was for putting that in core - my conditional is a superset of the original one, which means it will sometimes load an extra JS file.

Now that I'm knee-deep in it, I'll finish this up by changing the conditional and handling the other place this is included.

**Comment by MegaphoneJon** [comment]
@demeritcowboy  it came back to me.  I changed the conditional so that we should show a "this event is full" message regardless of whether there's a waitlist.  So that change you pointed out is intentional.  But is incomplete without having a message without the waitlist verbiage.  So I either need to cut the scope or expand it.  I'll consider that tomorrow.

**Comment by demeritcowboy** [comment]
I'll just close this one again but feel free to reopen.

**Pull request closed without merging**